hummingbot/client/config/config_helpers.py

```diff
@@ -28,11 +45,232 @@
     TRADE_FEES_CONFIG_PATH,
 )
 
-if TYPE_CHECKING:  # avoid circular import problems
-    from hummingbot.client.config.config_data_types import BaseStrategyConfigMap
-
 # Use ruamel.yaml to preserve order and comments in .yml file
-yaml_parser = ruamel.yaml.YAML()
+yaml_parser = ruamel.yaml.YAML()  # legacy
+
+
+def decimal_representer(dumper: SafeDumper, data: Decimal):
+    return dumper.represent_float(float(data))
+
+
+def enum_representer(dumper: SafeDumper, data: ClientConfigEnum):
+    return dumper.represent_str(str(data))
+
+
+def date_representer(dumper: SafeDumper, data: date):
+    return dumper.represent_date(data)
+
+
+def time_representer(dumper: SafeDumper, data: time):
+    return dumper.represent_str(data.strftime("%H:%M:%S"))
+
+
+def datetime_representer(dumper: SafeDumper, data: datetime):
+    return dumper.represent_datetime(data)
+
+
+yaml.add_representer(
+    data_type=Decimal, representer=decimal_representer, Dumper=SafeDumper
+)
+yaml.add_multi_representer(
+    data_type=ClientConfigEnum, multi_representer=enum_representer, Dumper=SafeDumper
+)
+yaml.add_representer(
+    data_type=date, representer=date_representer, Dumper=SafeDumper
+)
+yaml.add_representer(
+    data_type=time, representer=time_representer, Dumper=SafeDumper
+)
+yaml.add_representer(
+    data_type=datetime, representer=datetime_representer, Dumper=SafeDumper
+)
+
+
+class ConfigValidationError(Exception):
+    pass
+
+
+@dataclass()
+class ConfigTraversalItem:
+    depth: int
+    config_path: str
+    attr: str
+    value: Any
+    printable_value: str
+    client_field_data: Optional[ClientFieldData]
+    field_info: FieldInfo
+
+
+class ClientConfigAdapter:
+    def __init__(self, hb_config: BaseClientModel):
+        self._hb_config = hb_config
+
+    def __getattr__(self, item):
+        if item == "_hb_config":
+            value = super().__getattribute__(item)
+        else:
+            value = getattr(self._hb_config, item)
+            if isinstance(value, BaseClientModel):
+                value = ClientConfigAdapter(value)
+        return value
+
+    def __setattr__(self, key, value):
+        if key == "_hb_config":
+            super().__setattr__(key, value)
+        else:
+            try:
+                self._hb_config.__setattr__(key, value)
+            except ValidationError as e:
+                raise ConfigValidationError(retrieve_validation_error_msg(e))
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}.{self._hb_config.__repr__()}"
+
+    def __eq__(self, other):
+        if isinstance(other, ClientConfigAdapter):
+            eq = self._hb_config.__eq__(other._hb_config)
+        else:
+            eq = super().__eq__(other)
+        return eq
```

Comment:
I want to start pushing back on the insistence to use logical statements instead of if-else in cases like these, because I feel like the if-else form is more readable here